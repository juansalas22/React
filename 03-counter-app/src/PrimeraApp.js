/*
¡Ten en cuenta que nuestro código original ya no necesita importar React para usar JSX! 
(Pero aún necesitaríamos importar React para poder usar Hooks u otras exportaciones que
proporciona React).
*/
//import React, { Fragment } from 'react'; (Se importo para el Fragment)
import PropTypes from 'prop-types';
//FC
const PrimeraApp = ({saludo, parrafo}) => {

    return (
        // se puede utilizar <div></div> pero queda flotante
        // se puede utilizar <Fragment></Fragment> pero funciona con menos codigo:
        <> 
            <h1>{saludo}</h1>
            {/* <pre>{JSON.stringify(saludo, null, 3)}</pre> */}
            <p>{parrafo}</p>
        </>
        // EL <pre></pre> se usa para mostar un objeto
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}
PrimeraApp.defaultProps = {
    parrafo: "ahi les voy"
}

export default PrimeraApp;