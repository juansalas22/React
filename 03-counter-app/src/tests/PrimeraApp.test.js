import PrimeraApp from "../PrimeraApp";
import { shallow } from 'enzyme';



describe('Pruebas en <PrimeraApp />', () => {
    
 //    test('Debe de mostrar el mensaje "Hola, soy Goku', () => {
 //        const saludo = 'Hola, soy Goku';
 //        const {getByText} = render(<PrimeraApp saludo={saludo} />);
 //        expect(getByText(saludo)).toBeInTheDocument();
 //    })

    test('Debe mostar <PrimeraApp /> correctamente', () => {
        
        const saludo = 'Hola, soy Goku';
        const wrapper = shallow( <PrimeraApp saludo={saludo}/>);

        expect(wrapper).toMatchSnapshot();

    })

    test('Debe de mostrar el subtitulo enviado por props ', () => {
        const saludo = 'Hola, soy Goku';
        const subTitulo = 'ahi les voy';
        const wrapper = shallow(
             <PrimeraApp
                saludo={saludo}
                subtitulo={subTitulo}
              />
        );
        
        const textoParrafo = wrapper.find('p').text();
        
        expect(textoParrafo).toBe(subTitulo);

    })
    
    

})
