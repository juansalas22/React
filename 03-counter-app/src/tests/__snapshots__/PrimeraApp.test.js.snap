// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe mostar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    ahi les voy
  </p>
</Fragment>
`;
